File: t/PACSFramework/CS/MedicalWorkstationCustomizationDemo/Examples/PluggingViewExample/PlugWorkstationViewCommand.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Leadtools.Medical.Workstation.Commands;
using Leadtools.Medical.Workstation;
using Leadtools.Medical.Workstation.Interfaces.Views;
using Leadtools.Medical.Workstation.UI.Factory;

namespace Leadtools.Demos.Workstation.Customized
{
   class PlugWorkstationViewCommand <V,T>: WorkstationCommand where V : IWorkstationView
   {
      public PlugWorkstationViewCommand ( string featureId, WorkstationContainer container ) 
      : base ( featureId, container )
      {
         
      }
      
      protected override bool DoCanExecute()
      {
         return true ;
      }

      protected override void DoExecute()
      {
         WorkstationUIFactory.Instance.RegisterWorkstationView <V> ( typeof ( T) ) ;
      }
   }
}

File: t/PACSFramework/VB/MedicalWorkstationCustomizationDemo/Examples/PluggingViewExample/PlugWorkstationViewCommand.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports Leadtools.Medical.Workstation.Commands
Imports Leadtools.Medical.Workstation
Imports Leadtools.Medical.Workstation.Interfaces.Views
Imports Leadtools.Medical.Workstation.UI.Factory

Namespace Leadtools.Demos.Workstation.Customized
   Class PlugWorkstationViewCommand(Of V As IWorkstationView, T)
      Inherits WorkstationCommand
      Public Sub New(ByVal featureId As String, ByVal container As WorkstationContainer)
         MyBase.New(featureId, container)

      End Sub

      Protected Overrides Function DoCanExecute() As Boolean
         Return True
      End Function

      Protected Overrides Sub DoExecute()
         WorkstationUIFactory.Instance.RegisterWorkstationView(Of V)(GetType(T))
      End Sub
   End Class
End Namespace
